File: Exercism/dart/rna-transcription/lib/rna_transcription.dart

class RnaTranscription {
  var rna = {
    'C': 'G',
    'G': 'C',
    'T': 'A',
    'A': 'U',
  };

  String toRna(String input) {
    return input.split("").map((String s) => rna[s]).join("");
  }
}

File: Exercism/typescript/rna-transcription/rna-transcription.ts

class Transcriptor {
    toRna(dna: string): string {
        var rnas: { [val: string]: string } = {
            "C": "G",
            "G": "C",
            "T": "A",
            "A": "U",
        };

        var transcribe = (s: string) => {
            if (s in rnas)
                return rnas[s];
            throw Error('Invalid input DNA.');
        };

      // throw Error("At least two colors need to be present");
        return dna.split("").map(transcribe).join("");
    }
}

export default Transcriptor
